compiler/src/dotty/tools/dotc/reporting/diagnostic/ErrorKind.java

```diff
@@ -0,0 +1,28 @@
+package dotty.tools.dotc.reporting.diagnostic;
+
+public enum ErrorKind {
```

Comment:
It would be better to have this in Scala.

```scala
class ErrorKind(name: String) {
  def toString(): String = name
}
object ErrorKinds {
  val COMPATIBILITY = new ErrorKind("Compatibility")
  val DEFINITION_NOT_FOUND = new ErrorKind("Definition Not Found")
  ...
}
```

Reply:
We are already using java enums for ErrorMessageID. I think it is fine to stick to java enum until we migrate everything to Dotty enum.

I would however use Scala naming convention: `NoKind` instead of `NO_KIND`

Reply:
Ok.

But move `NoKind` to the first position.